### PR TITLE
Fix AdminServiceStore diesel migrations

### PR DESCRIPTION
**Pull request by agunde406**
Postgres updates were tested against a running postgres database and a test was added for running the sqlite migrations against an in memory sqlite database. Full unit tests against the sqlite database will be coming in future PR.

### CODE REVIEW

File: libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql [hunk 1]
```diff
@@ -73,7 +73,6 @@ CREATE TABLE IF NOT EXISTS proposed_service_argument (
     key                       TEXT NOT NULL,
     value                     TEXT NOT NULL,
     PRIMARY KEY (circuit_id, service_id, key),
-    FOREIGN KEY (service_id) REFERENCES proposed_service(service_id),
```
Comment (ltseeley) [inline libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql:75]: Seems to be a typo in the commit message: "... however this is not
service_id is not ..."

File: libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql [hunk 2]
```diff
@@ -113,6 +113,5 @@ CREATE TABLE IF NOT EXISTS circuit_member (
 CREATE TABLE IF NOT EXISTS node_endpoint (
     node_id                TEXT NOT NULL,
     endpoint               TEXT NOT NULL,
-    PRIMARY KEY (node_id, endpoint),
```
Comment (ltseeley) [inline libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql:117]: Typo in commit message: "... node_id is not unique in  and the circuit_member ...". Also s/delete/deleted/ at the end.

File: libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql (hunk 1 above)
Comment (agunde406) [reply libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql:75]: Fixed

File: libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql (hunk 2 above)
Comment (agunde406) [reply libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql:117]: fixed

**Review by ltseeley — DISMISSED** [review]

**Review by ltseeley — APPROVED** [review]

File: libsplinter/src/admin/store/diesel/mod.rs [hunk 3]
```diff
@@ -252,3 +252,34 @@ impl AdminServiceStore for DieselAdminServiceStore<diesel::sqlite::SqliteConnect
...
+
+    #[test]
+    fn test_sqlite_migrations() {
+        create_connection_pool_and_migrate();
+    }
+
+    /// Creates a conneciton pool for an in-memory SQLite database with only a single connection
+    /// available. Each connection is backed by a different in-memory SQLite database, so limiting
+    /// the pool to a single connection insures that the same DB is used for all operations.
+    fn create_connection_pool_and_migrate() -> Pool<ConnectionManager<SqliteConnection>> {
```
Comment (shannynalayna) [inline libsplinter/src/admin/store/diesel/mod.rs:273]: Some typos in the test comment here,  conneciton/connection, insures/ensures

Comment (agunde406) [reply libsplinter/src/admin/store/diesel/mod.rs:273]: Fixed the typos

**Review by shannynalayna — APPROVED** [review]

**Review by peterschwarz — APPROVED** [review]

**Pull request merged**